src/main/scala/io/iohk/ethereum/network/p2p/validators/OmmersValidator.scala

```diff
@@ -0,0 +1,138 @@
+    *
+    * @param blockNumber    The number of the block to which the ommers belong
+    * @param ommers         The list of ommers to validate
+    * @param blockchain     from where the ommer's parents will be obtained
+    * @return ommers if valid, an [[OmmersUsedBeforeError]] otherwise
+    */
+  private def validateOmmersAncestors(blockNumber: BigInt, ommers: Seq[BlockHeader], blockchain: Blockchain): Either[OmmersError, Seq[BlockHeader]] = {
+    val ancestors = getAncestors(blockNumber, OmmerGenerationLimit, blockchain)
+
+    val validOmmersAncestors: Seq[BlockHeader] => Boolean = anc =>
+      ommers.forall{ ommer =>
+        val ommerIsNotAncestor = anc.forall(_.hash != ommer.hash)
+        val ommersParentIsAncestor = anc.exists(_.parentHash == ommer.parentHash)
+        ommerIsNotAncestor && ommersParentIsAncestor
+      }
+    if(ancestors.exists(validOmmersAncestors(_))) Right(ommers)
+    else Left(OmmersAncestorsError)
+  }
+
+  /**
+    * Validates that each ommer was not previously used
+    * based on validations stated in the white paper (https://github.com/ethereum/wiki/wiki/White-Paper)
+    *
+    * @param blockNumber    The number of the block to which the ommers belong
+    * @param ommers         The list of ommers to validate
+    * @param blockchain     from where the ommer's parents will be obtained
+    * @return ommers if valid, an [[OmmersUsedBeforeError]] otherwise
+    */
+  private def validateOmmersNotUsed(blockNumber: BigInt, ommers: Seq[BlockHeader], blockchain: Blockchain): Either[OmmersError, Seq[BlockHeader]] = {
+    val usedOmmersHashes = getUsedOmmers(blockNumber, OmmerGenerationLimit, blockchain)
+
+    val allOmmersUnused: Seq[BlockHeader] => Boolean = usedOmmHashes =>
+      ommers.intersect(usedOmmHashes).isEmpty
+    if(usedOmmersHashes.exists(allOmmersUnused(_))) Right(ommers)
+    else Left(OmmersUsedBeforeError)
+  }
+
+  /**
+    * Validates that there are no duplicated ommers
+    * based on validations stated in the white paper (https://github.com/ethereum/wiki/wiki/White-Paper)
+    *
+    * @param ommers         The list of ommers to validate
+    * @return ommers if valid, an [[OmmersDuplicatedError]] otherwise
+    */
+  private def validateDuplicatedOmmers(ommers: Seq[BlockHeader]): Either[OmmersError, Seq[BlockHeader]] = {
+    if(ommers.distinct.length == ommers.length) Right(ommers)
+    else Left(OmmersDuplicatedError)
+  }
+
+  private def getAncestors(blockNumber: BigInt, numberOfBlocks: Int, blockchain: Blockchain): Option[Seq[BlockHeader]] = {
+    val ancestors: Seq[Option[BlockHeader]] = (1 to numberOfBlocks)
+      .collect{ case i if blockNumber - i >= 0 => blockNumber - i }
+      .map{ num => blockchain.getBlockHeaderByNumber(num) }
+    if(ancestors.exists(_.isEmpty)) None
+    else Some(ancestors.flatten)
+  }
+
+  private def getUsedOmmers(blockNumber: BigInt, numberOfBlocks: Int, blockchain: Blockchain): Option[Seq[BlockHeader]] = {
```

Comment:
minor comment. I would change the name of this method to something like `collectOmmersFromParents` or something like that